graph-builder/src/graph.rs

```diff
@@ -65,19 +65,26 @@ pub fn register_metrics(registry: &prometheus::Registry) -> Fallible<()> {
 }
 
 /// Serve Cincinnati graph requests.
-pub fn index(req: HttpRequest<State>) -> Result<HttpResponse, GraphError> {
+pub fn index(req: HttpRequest) -> Result<HttpResponse, GraphError> {
     V1_GRAPH_INCOMING_REQS.inc();
 
     // Check that the client can accept JSON media type.
     commons::ensure_content_type(req.headers(), CONTENT_TYPE)?;
 
     // Check for required client parameters.
-    let mandatory_params = &req.state().mandatory_params;
+    let mandatory_params = &req
+        .app_data::<State>()
+        .expect("the request has no app_data attached. this is a bug.")
```

Comment:
Here and similar code elsewhere, I am bit on the fence on this. On one hand, this shouldn't really happen, so a panic is justified. On the other hand, it could be easier to just handle this as another error variant, so that we can easily watch via metrics and alert on it.

Reply:
IMO this error shouldn't ever pass CI and I don't think it makes sense to track it.

Reply:
I think the new tests cases cover this one sufficiently.